src/plugins/intel_cpu/src/ngraph_transformations/convert_precision_i64_i32.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "convert_precision_i64_i32.hpp"
+#include <openvino/opsets/opset10.hpp>
+#include "transformations/utils/utils.hpp"
+#include "cpu_types.h"
+
+#include <unordered_set>
+
+bool isNativelySupported(const ov::Node::type_info_t &type) {
+    static const std::unordered_set<ov::Node::type_info_t> i64Ops = {
+        ov::opset10::Add::get_type_info_static(),
+        ov::op::v1::Broadcast::get_type_info_static(),
+        ov::opset10::Broadcast::get_type_info_static(),
+        ov::opset10::Constant::get_type_info_static(),
+        ov::opset10::Convert::get_type_info_static(),
+        ov::opset10::Divide::get_type_info_static(),
+        ov::opset10::Equal::get_type_info_static(),
+        ov::opset10::FloorMod::get_type_info_static(),
+        ov::op::v7::Gather::get_type_info_static(),
+        ov::opset10::Gather::get_type_info_static(),
+        ov::opset10::Greater::get_type_info_static(),
+        ov::opset10::Less::get_type_info_static(),
+        ov::opset10::Multiply::get_type_info_static(),
+        ov::opset10::NonMaxSuppression::get_type_info_static(),
+        ov::opset10::OneHot::get_type_info_static(),
+        ov::opset10::Parameter::get_type_info_static(),
+        ov::opset10::ReduceL1::get_type_info_static(),
+        ov::opset10::ReduceL2::get_type_info_static(),
+        ov::opset10::ReduceLogicalAnd::get_type_info_static(),
+        ov::opset10::ReduceMax::get_type_info_static(),
+        ov::opset10::ReduceMean::get_type_info_static(),
+        ov::opset10::ReduceMin::get_type_info_static(),
+        ov::opset10::ReduceProd::get_type_info_static(),
+        ov::opset10::ReduceSum::get_type_info_static(),
+        ov::opset10::Reshape::get_type_info_static(),
+        ov::opset10::Result::get_type_info_static(),
+        ov::opset10::ScatterElementsUpdate::get_type_info_static(),
+        ov::opset10::ScatterNDUpdate::get_type_info_static(),
+        ov::opset10::ScatterUpdate::get_type_info_static(),
+        ov::opset10::Select::get_type_info_static(),
+        ov::opset10::ShapeOf::get_type_info_static(),
+        ov::opset10::SquaredDifference::get_type_info_static(),
+        ov::opset10::Squeeze::get_type_info_static(),
+        ov::opset10::Subtract::get_type_info_static(),
+        ov::opset10::Tile::get_type_info_static(),
+        ov::opset10::Transpose::get_type_info_static(),
+        ov::opset10::Unique::get_type_info_static(),
+        ov::opset10::Unsqueeze::get_type_info_static()
```

Comment:
having such global maps does not allow conditional compilation feature to remove not used operations
We likely need to use CC factory here

Reply:
Yes I think linker won;t be able to remove unused operations for static build if we have references here.